go/database/mpt/state_test.go

```diff
@@ -341,6 +341,50 @@ func TestState_StateModifications_Failing(t *testing.T) {
 	}
 }
 
+func TestState_HasEmptyStorage(t *testing.T) {
+	for name, open := range mptStateFactories {
+		t.Run(name, func(t *testing.T) {
+			t.Run(name, func(t *testing.T) {
+				dir := t.TempDir()
+
+				state, err := open(dir)
+				if err != nil {
+					t.Fatalf("cannot open state: %s", err)
+				}
+
+				addr := common.Address{0x1}
+				injectedErr := errors.New("injected error")
+				ctrl := gomock.NewController(t)
+				db := NewMockDatabase(ctrl)
+				db.EXPECT().HasEmptyStorage(gomock.Any(), addr).Return(true, nil)
```

Comment:
I think no need to test all possible outputs as long as it still test the same controlflow.